### PR TITLE
gar migration

### PR DESCRIPTION
**Pull request by anwilkie**
# What and why?

This PR is to migrate from google container registry to google artifact registry.

# How to test?
Deploy this PR to your environment, setting the configBranch to `ras-1373-gar-migration`. The build will fail initially, but you can fix it by editing the image variable in the sample deployment's YAML file to `europe-west2-docker.pkg.dev/ons-ci-rmrasbs/images/sample:pr-260`.

# Jira
[Card](https://jira.ons.gov.uk/browse/RAS-1373)

### CODE REVIEW

**Review by matthew-robinson-ons — APPROVED** [review]
Can see pr image in GAR and it deploys and tests all pass. I had to update my docker config using `gcloud auth configure-docker europe-west2-docker.pkg.dev
` to get the collex and party images to be pulled when building locally.

File: _infra/helm/sample/values.yaml [hunk 1]
```diff
@@ -2,8 +2,8 @@ env: minikube
 namespace: minikube
 
 image:
-  devRepo: eu.gcr.io/ons-rasrmbs-management
-  name: eu.gcr.io/ons-rasrmbs-management
+  devRepo: europe-west2-docker.pkg.dev/ons-ci-rmrasbs/images
+  name: europe-west2-docker.pkg.dev/ons-ci-rmrasbs/images
```
Comment (pricem14pc) [inline _infra/helm/sample/values.yaml:6]: Not for this PR as its the same across all our charts, but I've never fully understood `devRepo` vs `name` for these two.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -9,8 +9,10 @@ on:
     branches: [ main ]
 env:
   IMAGE: sample
-  REGISTRY_HOSTNAME: eu.gcr.io
-  HOST: ${{ secrets.GOOGLE_PROJECT_ID }}
+  REGISTRY_HOSTNAME: europe-west2-docker.pkg.dev
+  SPINNAKER_GOOGLE_PROJECT_ID: ${{ secrets.SPINNAKER_GOOGLE_PROJECT_ID }}
+  GAR_REPOSITORY: images
+  GAR_GOOGLE_PROJECT_ID: ${{ secrets.GAR_GOOGLE_PROJECT_ID }}
   RELEASE_HOST: ${{ secrets.RELEASE_PROJECT_ID }}
```
Comment (pricem14pc) [inline .github/workflows/main.yml:16]: Is `RELEASE_HOST` now redundant? If so this should be removed.

Comment (anwilkie) [reply .github/workflows/main.yml]: Yes, it's redundant. I've removed it now.

**Review by pricem14pc — APPROVED** [review]
PR Image building and is pushed to GAR. Other references look good.

**Pull request merged**